### ISSUE TITLE
Add compatibility with Nextcloud 22

### ISSUE THREAD

**Issue by llucax** [ayselafsar/dicomviewer#76]
**Describe the bug**
Related to #68, this app is not marked as compatible with Nextcloud 19.

**To Reproduce**
1. Install Nextcloud 19
2. Try to install this app

**Expected behavior**
It is installed and enabled and works.

**Screenshots**
![image](https://user-images.githubusercontent.com/1031485/91663137-44c00000-eae7-11ea-9fbd-c8814c157313.png)

**Client details:**
 - OS: Linux
 - Browser All
 - Version N/A
 - Device: All

<details>
<summary>Server details</summary>

**DICOM viewer app version:** 1.2.2

**Operating system**: Linux

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.3

**Nextcloud version:** 19.0.2

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
Not relevant

#### Browser log
Not relevant

</details>

**Comment by stratege1401** [comment]
+1 !!!

**Comment by ciroiriarte** [comment]
Holding back on Nextcloud 19 because of this, although it should be tagged as enhancement request, not bug

**Comment by staffa007** [comment]
+1!!

**Comment by BernieO** [comment]
I just updated two instances to Nextcloud 19.0.3 and the dicomviewer still works.
Of course I can't give a guarantee that dicomviewer will work with your setups after upgrade to v19, so
be sure to have a backup of Nextcloud in case you have to revert the update.

**Comment by arnowelzel** [comment]
I can confirm that DICOM viewer works with Nextcloud 19.0.3 and 19.0.4.

**Comment by crami** [comment]
Can also confirm that it works with 19.0.4

**Comment by BernieO** [comment]
Out of topic for this issue actually, but posting it anyway:
in my instance the dicom-viewer also works with Nextcloud 20.0.1

**Comment by dumblob** [comment]
Yeah, it's off topic, but there is Nextcloud 20 already, so maybe this issue should be renamed?

**Comment by Apieceofbluesky** [comment]
In my Nextcloud 20 Installation, the viewer started to load, but got stuck on 100% and never actually displayed the DICOM file. I'm not sure if it's a compatibility or general issue.

**Comment by Numrollen** [comment]
So, the app is dead? Its still advertising  https://nextcloud.com/industries/healthcare/

And im in hospital, showed the app as a nice addon. Now we start with NC. Nice....

**Comment by ayselafsar** [comment]
Hi everyone,

Sorry for my late reply! I'm currently working on another exciting open source Nextcloud application that will be released soon, so I couldn't work on DICOM Viewer due to my limited availability. However, I'm planning a major release with new features for DICOM Viewer in Q1 2021. In the meantime, I would be happy to review and accept any pull requests.

Please stay tuned and thanks for your patience!

**Comment by arnowelzel** [comment]
I can confirm, that DICOM viewer works in a fresh installation of Nextcloud 20.0.2 without any major issues. There are just two error messages in the developer console of Firefox:

```
downloadable font: Failed to convert WOFF 2.0 font to SFNT (font-family: "FontAwesome" style:normal weight:400 stretch:100 src index:1) source: https://server.example/apps/dicomviewer/css/external/font-awesome/fonts/fontawesome-webfont.woff2
downloadable font: rejected by sanitizer (font-family: "FontAwesome" style:normal weight:400 stretch:100 src index:1) source: https://server.example/apps/dicomviewer/css/external/font-awesome/fonts/fontawesome-webfont.woff2
```

Looks like a missing filter for font files in Webpack or whatever you use to build the UI. However, the viewer itself seems to work fine nevertheless and so far I did not miss any icon or symbol in the UI.

When updating the app, you should also remove all the macOS specific artifacts like `._.DS_Store`, `.DS_Store` and all other files starting with a dot.

**Comment by arnowelzel** [comment]
> 
> 
> In my Nextcloud 20 Installation, the viewer started to load, but got stuck on 100% and never actually displayed the DICOM file. I'm not sure if it's a compatibility or general issue.

In my test environment with Nextcloud 20.0.2 the viewer works without any problem. So there must be some issue with your setup. Did you try clearing the cache of your browser or maybe another browser just to be sure?

**Comment by arnowelzel** [comment]
> 
> 
> Hi everyone,
> 
> Sorry for my late reply! I'm currently working on another exciting open source Nextcloud application that will be released soon, so I couldn't work on DICOM Viewer due to my limited availability. However, I'm planning a major release with new features for DICOM Viewer in Q1 2021. In the meantime, I would be happy to review and accept any pull requests.
> 
> Please stay tuned and thanks for your patience!

Thanks for the feedback! Excellent work!

**Comment by tuxmaster5000** [comment]
Today I have updated to 20.0.5 and it will works.

**Comment by stratege1401** [comment]
works on 21.0.1

**Comment by Numrollen** [comment]
> Hi everyone,
> 
> Sorry for my late reply! I'm currently working on another exciting open source Nextcloud application that will be released soon, so I couldn't work on DICOM Viewer due to my limited availability. However, I'm planning a major release with new features for DICOM Viewer in Q1 2021. In the meantime, I would be happy to review and accept any pull requests.
> 
> Please stay tuned and thanks for your patience!

So, any News on this one?

**Comment by ciroiriarte** [comment]
I'm on 22, it installs and seems to work :)

**Comment by arnowelzel** [comment]
@ayselafsar If you need any help with testing the package so it can at least be used with Nextcloud 21/22, let me know. I am a co-maintainer of https://github.com/jhass/nextcloud-keeweb/ and also use your application here. At the moment I think it would be ok just to update the supported Nextcloud version number since the application itself seems to work fine in Nextcloud 21/22 as well.

**Comment by ciroiriarte** [comment]
I've updated to nextcloud 22.2.0.2, it still works.

The only annoyance is that there's a delay loading studies. Cannot frankly recall if it has always being like that, but I got to notice it now.

**Comment by Numrollen** [comment]
thx for your effort. Im still hoping for a new supported release, perhaps even with improvments..

**Comment by Lars1963** [comment]
@ayselafsar As NC23 is about to be released (scheduled for december 1st), you might add compatibilty for that as well. 🤔

**Comment by ayselafsar** [comment]
Fixed on v1.2.3

Thanks for your patience!
